collapse.go

```diff
@@ -23,7 +23,7 @@ func processRewardWithdrawals(round uint64, snapshot *avl.Tree) {
 	}
 }
 
-func rewardValidators(g *Graph, snapshot *avl.Tree, root *Transaction, tx *Transaction, logging bool) error {
+func rewardValidators(g *Graph, snapshot *avl.Tree, tx *Transaction, logging bool) error {
```

Comment:
so are we passing by value or by pointer? i though that idea is to pass values everywhere when we do not need explicitly mutate state to prevent potential errors.
in this particular case, i guess we can pass only data used inside, instead of whole transaction

Reply:
For this case we pass it by pointer because we can assume that all transactions called with rewardValidators() must be stored within the graph, and that all functions that call rewardValidators() already holds a reference to the pointer to the tx transaction.

Reply:
if graph has all transaction then passing tx.ID should be sufficient right? because you can take it's content from the graph.

Reply:
That is true, the point is to prevent a lookup to optimize performance since rewardValidators() by itself is already very expensive. Will document it.